docs/module.md

```diff
@@ -271,6 +271,52 @@ Those scripts will be made accessible like any other command in the task environ
 This feature requires the use of a local or shared file system for the pipeline work directory, or {ref}`wave-page` when using cloud-based executors.
 :::
 
+## Module config
+
+:::{versionadded} 23.11.0-edge
+:::
+
+Modules can define a config file called `module.config` in the module directory. This config file can be used to apply process configuration to processes that are invoked in the module. Unlike a regular Nextflow configuration file, the module config assumes the {ref}`process config scope <config-process>` and only supports process config settings.
+
+Here is an example module with a module config:
+
+```
+<module-dir>
+|── main.nf
+└── module.config
+```
+
+```groovy
+// main.nf
+process BAR {
+}
+
+workflow FOO {
+    BAR()
+}
+```
+
+```groovy
+// module.config
+withName:'FOO:BAR' {
```

Comment:
Questions:
- what happens in this PR when using `withName: BAR` in the module config, without the `FOO:`?
- what happens in this PR when defining module config lines with no Name/Label selectors?

(asking from a user experience perspective - haven't looked at the code nor tested yet, on purpose)

Reply:
The standard priorities apply, from lowest to highest:
- generic process config
- process definition
- withLabel
- withName (base name)
- withName (imported name)
- withName (fully qualified name)